fvm/scriptEnv.go

```diff
@@ -685,3 +686,35 @@ func (e *ScriptEnv) Commit() ([]programs.ContractUpdateKey, error) {
 func (e *ScriptEnv) AllocateStorageIndex(_ []byte) (uint64, error) {
 	return 0, errors.NewOperationNotSupportedError("AllocateStorageIndex")
 }
+
+func (e *ScriptEnv) BLSVerifyPOP(pk *runtime.PublicKey, sig []byte) (bool, error) {
+	key, err := crypto.DecodePublicKey(crypto.BLSBLS12381, pk.PublicKey)
+	if err != nil {
+		return false, err
+	}
+	return crypto.VerifyPOP(key, sig)
+}
+
+func (e *ScriptEnv) AggregateBLSSignatures(sigs [][]byte) ([]byte, error) {
+	return crypto.AggregateSignatures(sigs)
+}
+
+func (e *ScriptEnv) AggregateBLSPublicKeys(keys []*runtime.PublicKey) (*runtime.PublicKey, error) {
+	pks := make([]crypto.PublicKey, 0, len(keys))
+	for _, key := range keys {
+		pk, err := crypto.DecodePublicKey(crypto.BLSBLS12381, key.PublicKey)
+		if err != nil {
+			return nil, err
+		}
+		pks = append(pks, pk)
+	}
+	pk, err := crypto.AggregatePublicKeys(pks)
```

Comment:
Same here, we probably should check for the sentinel error (we can make the check under fvm/crypto)